edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/Avaliability.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
General comment: should we move these generic metric related class in another new project in order to share across projects, e.g. edgeAgent and edgeHub.

Reply:
Availability seems pretty specific to edge agent. Is there any availability that hub would be measuring? maybe network availability?